### PR TITLE
Implement JOSS doc suggestions

### PR DESCRIPTION
**Pull request by jschepers**
We implemented the doc suggestions from the JOSS review and fixed/modified some things on the way.

## Changes related to the review:
* **Quickstart page:** We added a short introduction about which ingredients are needed for a simulation. Hopefully, the given tip is more useful now given the context,

* **Simulate ERPs page:** We added an introductory paragraph to explain the contexts of ERP components and give an example.

* **Poweranalysis page:** We split "poweranalysis" in two words and matched sidebar and page title.

* **Reference Section:**
  * We changed the word in the README from "component" to "simulation ingredient" to avoid the overloading of "component".
  * We matched the sub-titles from the sidebar and the titles of the pages.
  * We reordered the sub-sections of the reference section to match the order from the overview page.
  * We added a page about "Experimental design types" with a description about the design types that are currently implemented.

* **API/DocStrings:**
  * We removed some unintended empty lines between the `simulate()` docstring and the function such that it is recognized as a doc string.
  * We added some information to the `SingleSubjectDesign()` and the `MultiSubjectDesign()` docstrings.
  * We fixed the formatting of the `predef_eeg()` doc string.
  * We added a reference for the different colours of noise on the reference page "Overview: Noise types".
  * We corrected the spelling of "Artifact.jl" in the `RealisticNoise()` docstring
  * We added more explanation to the `ExponentialNoise()` docstring.

* **General changes**
  * We tried to match all sub-title in the sidebar with the page titles.
  * We added an introductory sentence to all pages.

## Additional changes:
* We added the UnfoldSim logo to the documentation page.
* We wrapped the package imports in collapsible setup blocks.

### CODE REVIEW

File: docs/literate/reference/noisetypes.jl [hunk 1]
```diff
@@ -1,10 +1,16 @@
+# # Overview: Noise types
+
+# There are different types of noise signals which differ in their power spectra.
+# If you are not familiar with different types/colors of noise yet, have a look at this [source](https://en.wikipedia.org/wiki/Colors_of_noise).
```
Comment (behinger) [inline docs/literate/reference/noisetypes.jl:4]:
```suggestion
# If you are not familiar with different types/colors of noise yet, have a look at the[colored noise wikipedia page](https://en.wikipedia.org/wiki/Colors_of_noise).
```

File: docs/literate/reference/overview.jl [hunk 2]
```diff
@@ -1,7 +1,17 @@
 # # Overview of functionality
 # UnfoldSim has many modules, here we try to collect them to provide you with an overview.
```
Comment (behinger) [inline docs/literate/reference/overview.jl:2]:
```suggestion
# A UnfoldSim simulation has four ingredients: Design, Component, Onset and Noise. Here we provide a short overview of the implemented functions.
```

**Comment by behinger** [comment]
very nice & well done. These are the only two things I noticed

File: docs/literate/reference/noisetypes.jl [hunk 3]
```diff
@@ -1,10 +1,16 @@
+# # Overview: Noise types
+
+# There are different types of noise signals which differ in their power spectra.
+# If you are not familiar with different types/colors of noise yet, have a look at the[colored noise wikipedia page](https://en.wikipedia.org/wiki/Colors_of_noise).
```
Comment (jschepers) [inline docs/literate/reference/noisetypes.jl:4]:
```suggestion
# If you are not familiar with different types/colors of noise yet, have a look at the[colors of noise Wikipedia page](https://en.wikipedia.org/wiki/Colors_of_noise).
```

File: docs/literate/reference/overview.jl [hunk 4]
```diff
@@ -1,7 +1,17 @@
 # # Overview of functionality
-# UnfoldSim has many modules, here we try to collect them to provide you with an overview.
+# A UnfoldSim simulation has four ingredients: Design, Component, Onset and Noise. Here we provide a short overview of the implemented functions.
```
Comment (jschepers) [inline docs/literate/reference/overview.jl:2]:
```suggestion
# A UnfoldSim simulation has four ingredients: Design, Component, Onset and Noise. Here we provide a short overview of the implemented types.
```

**Comment by jschepers** [comment]
Dear @roualdes,

thanks again for your detailed list of suggestions. They were really helpful. We revised the documentation accordingly (see the description above). You can find a preview of the changes here: https://unfoldtoolbox.github.io/UnfoldSim.jl/previews/PR101/
Please let us know if your suggestions were sufficiently addressed or if anything is still missing.

Best,
Judith

**Pull request merged**